plugins/inputs/dcos_containers/dcos_containers.go

```diff
@@ -0,0 +1,310 @@
+package dcos_containers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"math"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	"github.com/mesos/mesos-go/api/v1/lib"
+	"github.com/mesos/mesos-go/api/v1/lib/agent"
+	"github.com/mesos/mesos-go/api/v1/lib/agent/calls"
+	"github.com/mesos/mesos-go/api/v1/lib/httpcli"
+	"github.com/mesos/mesos-go/api/v1/lib/httpcli/httpagent"
+)
+
+const sampleConfig = `
+## The URL of the local mesos agent
+mesos_agent_url = "http://$NODE_PRIVATE_IP:5051"
+## The period after which requests to mesos agent should time out
+timeout = 10s
+`
+
+// DCOSContainers describes the options available to this plugin
+type DCOSContainers struct {
+	MesosAgentUrl string
+	Timeout       time.Duration
+}
+
+// measurement is a combination of fields and tags specific to those fields
+type measurement struct {
+	name   string
+	fields map[string]interface{}
```

Comment:
Why `fields` values are `interface{}` it looks like we want to keep there only numeric values is this correct? Why not simply use `float64`

Reply:
Thanks for looking over my PR! `fields` are values returned from getters on the [mesos proto](https://github.com/mesos/mesos-go/blob/master/api/v1/lib/mesos.pb.go). Those values can be `uint64`, `uint32`, or `float64`. 

Since they are fed into the `telegraf.Accumulator` which expects `map[string]interface{}`: https://github.com/dcos/telegraf/blob/1.7.2-dcos/accumulator.go#L14-L17, it makes sense to store them as `map[string]interface{}` in the interim.